completion.go

```diff
@@ -73,35 +73,51 @@ func (c *completion) skipPositional(s *parseState, n int) {
 	}
 }
 
-func (c *completion) completeOptionNames(names map[string]*Option, prefix string, match string) []Completion {
-	n := make([]Completion, 0, len(names))
+func (c *completion) completeOptionNames(s *parseState, prefix string, match string, short bool) []Completion {
+	if short && len(match) != 0 {
+		return []Completion{
+			Completion{
+				Item: prefix + match,
+			},
+		}
+	}
 
-	for k, opt := range names {
-		if strings.HasPrefix(k, match) && !opt.Hidden {
-			n = append(n, Completion{
-				Item:        prefix + k,
+	var results []Completion
+	repeats := map[string]bool{}
+
+	for name, opt := range s.lookup.longNames {
+		if strings.HasPrefix(name, match) && !opt.Hidden {
+			results = append(results, Completion{
+				Item:        "--" + name,
 				Description: opt.Description,
 			})
+
+			if short {
+				repeats[string(opt.ShortName)] = true
+			}
 		}
 	}
 
-	return n
-}
+	if short {
+		for name, opt := range s.lookup.shortNames {
+			if _, exist := repeats[name]; !exist && strings.HasPrefix(name, match) && !opt.Hidden {
+				results = append(results, Completion{
+					Item:        "-" + name,
```

Comment:
This should use the correct platform dependent short option prefix (`defaultShortOptDelimiter`)